src/main/java/org/gradle/profiler/BuildMutatorFactory.java

```diff
@@ -13,13 +13,16 @@ public BuildMutatorFactory(List<? extends Supplier<BuildMutator>> factories) {
 
     @Override
     public BuildMutator get() {
-        if (factories.isEmpty()) {
-            return new NoOpMutator();
+        List<BuildMutator> mutators = factories.stream()
+            .map(Supplier::get)
+            .filter(it -> it != BuildMutator.NOOP)
+            .collect(Collectors.toList());
+        if (mutators.isEmpty()) {
+            return BuildMutator.NOOP;
         }
-        if (factories.size() == 1) {
-            return factories.get(0).get();
+        if (mutators.size() == 1) {
+            return mutators.get(0);
         }
-        List<BuildMutator> mutators = factories.stream().map(Supplier::get).collect(Collectors.toList());
         return new CompositeBuildMutator(mutators);
```

Comment:
Nit: you could switch on the size of `mutators`.

Reply:
Done.